ambry-account/src/main/java/com/github/ambry/account/mysql/DatasetDao.java

```diff
@@ -918,6 +959,47 @@ private Page<String> executeListAllValidDatasetsStatement(PreparedStatement stat
     }
   }
 
+  /**
+   * Execute listValidDatasetVersionsSql statement.
+   * @param statement the listValidDatasetVersionsSql statement.
+   * @param accountId the id for the parent account.
+   * @param containerId the id of the container.
+   * @param datasetName the name of the dataset.
+   * @param pageToken the page token.
+   * @param versionSchema the dataset version schema.
+   * @return the page of the dataset versions.
+   * @throws SQLException
+   */
+  private Page<String> executeListAllValidDatasetVersionsStatement(PreparedStatement statement, int accountId,
+      int containerId, String datasetName, String pageToken, Dataset.VersionSchema versionSchema) throws SQLException {
+    ResultSet resultSet = null;
+    try {
+      statement.setInt(1, accountId);
+      statement.setInt(2, containerId);
+      statement.setString(3, datasetName);
+      statement.setInt(4, DatasetVersionState.READY.ordinal());
+      statement.setString(5, pageToken);
+      statement.setString(6, pageToken);
+      statement.setInt(7, mySqlAccountServiceConfig.listDatasetVersionsMaxResult + 1);
+      resultSet = statement.executeQuery();
+      List<String> entries = new ArrayList<>();
+      String nextContinuationToken = null;
+      int resultIndex = 0;
+      while (resultSet.next()) {
+        long versionValue = resultSet.getLong(VERSION);
+        String version = convertVersionValueToVersion(versionValue, versionSchema);
+        if (resultIndex++ == mySqlAccountServiceConfig.listDatasetVersionsMaxResult) {
```

Comment:
nit: I really don't like have `++` operator right after. can you increase the resultIndex right after this if statement.

Reply:
?
```
if (resultIndex == mysqlAccountServiceConfig.listDatasetVersionsMaxResult) {
    nextContinuationToken = version;
    break;
}
resultIndex ++;
```
This is exactly the same as your PR since you are not using resultIndex within the if statement and after while loop.

Reply:
You are right, updated.